README.md

```diff
@@ -92,158 +76,166 @@ Additionally, the device and any additional complementary hardware components sh
+   Valid options are `SD`, `SPIF`, `QSPIF` and `FLASHIAP` (not recommended). For more available options, please see the [block device components](https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice).
+
+   You also have to specify block device pin configuration, which may vary from one block device type to another. Here's an example for `SD`:
+      
    ```
             "sd.SPI_MOSI"                      : "PE_6",
             "sd.SPI_MISO"                      : "PE_5",
             "sd.SPI_CLK"                       : "PE_2",
             "sd.SPI_CS"                        : "PE_4",
    ```
    
-   * **Flash** - define the basics for the microcontroller flash, e.g.:
+- **Flash** - Define the basics for the microcontroller flash. For example:
+   
    ```
             "device_management.flash-start-address"              : "0x08000000",
             "device_management.flash-size"                       : "(2048*1024)",
    ```
    
-   * **SOTP** - define 2 SOTP/NVStore regions which will be used for Mbed OS Device Management to store it's special keys which are used to encrypt the data stored on the storage. Use the last 2 Flash sectors (if possible) to ensure that they don't get overwritten when new firmware is applied. Example:
+- **SOTP** - Define two SOTP or NVStore regions that Mbed OS Device Management will use to store its special keys, which encrypt the data stored. Use the last two Flash sectors (if possible) to ensure that they don't get overwritten when new firmware is applied. For example:
+
    ```
             "device_management.sotp-section-1-address"            : "(MBED_CONF_APP_FLASH_START_ADDRESS + MBED_CONF_APP_FLASH_SIZE - 2*(128*1024))",
             "device_management.sotp-section-1-size"               : "(128*1024)",
             "device_management.sotp-section-2-address"            : "(MBED_CONF_APP_FLASH_START_ADDRESS + MBED_CONF_APP_FLASH_SIZE - 1*(128*1024))",
             "device_management.sotp-section-2-size"               : "(128*1024)",
             "device_management.sotp-num-sections" : 2
    ```
-   `*-address` defines the start of the Flash sector and `*-size` defines the actual sector size. Currently `sotp-num-sections` should always be set to `2`.
 
-At this point, it's recommended that you run the "connect" test suite, which verifies that the device can successfully bootstrap, register and send/receive data from Pelion Device Management service with the provided configuration.
+`*-address` defines the start of the Flash sector, and `*-size` defines the actual sector size. `sotp-num-sections` should always be set to `2`.
+
+At this point, we recommend you run the "connect" test suite, which verifies that the device can successfully bootstrap, register and send and receive data from Pelion Device Management service with the provided configuration.
```

Comment:
Is this the first time we mention the connect test suite? Can we add a link or more clarification?